jax/_src/numpy/lax_numpy.py

```diff
@@ -1805,7 +1805,8 @@ def _split(op, ary, indices_or_sections, axis=0):
   size = ary.shape[axis]
   if isinstance(indices_or_sections, (tuple, list) + _arraylike_types):
     indices_or_sections = np.array(
-        [core.concrete_or_error(np.int64, i_s, f"in jax.numpy.{op} argument 1")
+        [_canonicalize_axis(core.concrete_or_error(np.int64, i_s,
+                                                   f"in jax.numpy.{op} argument 1"), size)
```

Comment:
Unrelated to your change, but the explicit cast to int64 here allows non-integer floats to be passed as indices. We should probably use `operator.index` rather than `np.int64`.

Reply:
I see. Looking at `testSplitTypeError` in `lax_numpy_test.py`, which calls `jnp.split` on some floating point values, I assumed that was intended. Indeed, changing to `operator.index` breaks that test for me locally (the last JVP evaluation, specifically). Can you even evaluate a JVP of an integer-domain function?

Reply:
Thanks for looking into it! It's not clear to me that the JVP over split indices is meaningful, but we can probably punt on that for now.